### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rohjs**
## Description 

- [x] Create a `CoingeckoService`
- [x] Create custom hooks that fetch data

### CODE REVIEW

File: lib/graphql.ts [hunk 1]
```diff
@@ -17,16 +17,21 @@ export const client = new ApolloClient({
...
-// NOTE: https://thegraph.com/hosted-service/subgraph/balancer-labs/balancer
-export const POOL_TOKEN_BALANCES = gql`
-  {
-    pool(
-      id: "${BPT_POOL_ID}"
-    ) {
-      tokens {
-        symbol
-        balance
+export async function fetchPoolTokenBalances() {
```
Comment (BasixKOR) [inline lib/graphql.ts:6]: If we don't have any usage for Apollo elsewhere, you may want to consider `graphql-request` as a more lightweight solution.

File: lib/coingecko.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  async getTokens(symbols: string[]) {
+    try {
+      const results = await this.fetchTokens(symbols)
+      const priceMap = results.reduce((acc, result) => {
+        Object.entries(result.data).forEach(([key, value]) => {
+          const symb = this.getSymbolName(key)
+          acc[symb] = value.usd
+        })
+        return acc
+      }, {} as PriceMap)
```
Comment (BasixKOR) [inline lib/coingecko.ts]: You may want to use [`Object.fromEntries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) API here.

**Comment by rohjs** [comment]
@BasixKOR I've made changes as requested! Thank you :)

**Review by BasixKOR — APPROVED** [review]
Still have a bit of mixed feelings with a lot of usePrevious hooks, but I think we're good to go.

**Pull request merged**